Storage/container/many_to_many.h

```diff
@@ -110,24 +108,27 @@ class GenericMatrix {
   }
   void operator()(const DELETE_EVENT& e) { DoErase(std::make_pair(e.key.first, e.key.second)); }
 
+  template <typename MAP>
```

Comment:
This generic Iterator class is too generic for `Storage/`, IMHO, how about moving it into `Bricks/util/iterator.h` (and adding a small test for it too)?